utils/bot/logger/config.go

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package logger
+
+type Config struct {
+	// AdminLogLevel is "debug", "info", "warn", or "error".
+	AdminLogLevel string
+
+	// AdminLogVerbose: set to include full context with admin log messages.
+	AdminLogVerbose bool
+}
+
+func (c Config) ToStorableConfig(configMap map[string]interface{}) map[string]interface{} {
+	if configMap == nil {
+		configMap = map[string]interface{}{}
+	}
+	configMap["AdminLogLevel"] = c.AdminLogLevel
+	configMap["AdminLogVerbose"] = c.AdminLogVerbose
```

Comment:
I did not dive much into the Bot part, since I was mainly re-adapting what was already done on MSCalendar. We could consider defining it as map[string]string since it is what we use now, but if we wanted to extend the Logger with other context that accept, for example, boolean, we would have to codify those as strings.

Reply:
I meant could the `configMap` be a struct, but if it needs to be dynamic per plugin then I suppose not.